CharityDB/1 tables.sql

```diff
@@ -0,0 +1,17 @@
+use CharityDB
+go
+drop table if exists Donor
+go 
+
+create table dbo.Donor(
+    DonorId int not null identity primary key,
+    FirstName varchar(25) not null constraint ck_Donor_FirstName_cannot_be_blank check(FirstName <> ''),
+    LastName varchar(25) not null constraint ck_Donor_LastName_cannot_be_blank check(LastName <> ''),
+    Address varchar(50) not null constraint ck_Donor_Address_cannot_be_blank check(Address <> ''),
+    PostalCode varchar(10) not null constraint ck_Donor_PostalCode_cannot_be_blank check(PostalCode <> ''),
+    DonationAmount decimal(6,2) not null constraint ck_Donor_DonationAmount_must_be_between_1_and_5000 check(DonationAmount between 1 and 5000),
+    Year int not null constraint ck_Donor_Year_must_be_between_2018_and_the_current_year check(Year between 2018 and year(getdate())),
+    Season varchar(6) not null constraint ck_Donor_Season_must_be_autumn_or_spring check(Season in('Autumn', 'Spring')),
```

Comment:
So I assume that means each donor must be unique per year, season. For the donor for this uinque constraint I used their address since you said the address must be unique so it is the same thing as the person themself.

Reply:
Thanks, though please change it to season and not year, they can donate twice a year, once per mailing.

Reply:
- If you make a constraint that it is just unique per season that a donor will only be able to donate twice all together, once for the season of autumn and once for spring. A constraint on person, season, year allows them to donate twice a year.
- One unique does not cover for the other, if you only have the second constraint with all 4 than it would not block there being the same address twice to one Postal code.

Reply:
I have come to realize that you cannot make FirstName and LastName unique alone, or Address and PostalCode...because you have the same donors repeated many times and therefore it will crash. So you need 2 constraints: 

1. FirstName, LastName, Year, Season
2. Address, PostalCode, Year, Season
Unless you want them unique all together, I can switch it to one constraint.
Your data does not follow all the rules you just added...so it crashes.

Reply:
What is the benefit of having it as one or two constraints?  That's correct, it seems like bad data, but those are the constraints I wanted. i have amended the data, so it no longer crashes.

Reply:
It is 2 separate things, the way a unique constraint works is that whatever columns you include those columns are only unique with eachother. So if you make a constraint on all of them together it is a much more lose constraint since they are only unique if all 6 are together. 2 separate constraints is tighter.